src/Lumper.UI/App.axaml.cs

```diff
@@ -17,11 +18,22 @@ public override void OnFrameworkInitializationCompleted()
     {
         if (ApplicationLifetime is IClassicDesktopStyleApplicationLifetime
             desktop)
+        {
+            var vm = new MainWindowViewModel();
             desktop.MainWindow = new MainWindow
             {
-                DataContext = new MainWindowViewModel()
+                DataContext = vm,
             };
 
+            desktop.MainWindow.AddHandler(DragDrop.DragOverEvent, MainWindowViewModel.DragOver);
+            desktop.MainWindow.AddHandler(DragDrop.DropEvent, vm.Drop);
+
+            if (desktop.Args?.Length > 0)
+            {
+                vm.LoadBsp(desktop.Args[0]);
+            }
```

Comment:
This can be in the MainWindowViewModel so you don't need to create a new ViewModel. There is a `desktop` already in the constructor in `MainWindowViewModel.cs` but maybe it should be in `MainWindowViewModel.IO.cs`. But we might add different `desktop.Args` in the future though so your choice.

Reply:
I'll pull this into VM for now. There's room for more robust argument parsing in a different PR.